std/variant.d

```diff
@@ -74,28 +74,48 @@ import std.meta, std.traits, std.typecons;
 
 /++
     Gives the `sizeof` the largest type given.
+
+    See_Also: https://forum.dlang.org/thread/wbpnncxepehgcswhuazl@forum.dlang.org?page=1
   +/
-template maxSize(T...)
+template maxSize(Ts...)
 {
-    static if (T.length == 1)
-    {
-        enum size_t maxSize = T[0].sizeof;
-    }
-    else
-    {
-        import std.algorithm.comparison : max;
-        enum size_t maxSize = max(T[0].sizeof, maxSize!(T[1 .. $]));
+    align(1) union Impl {
+        static foreach (i, T; Ts) {
+            static if (!is(T == void))
+                mixin("T _field_", i, ";");
+        }
     }
+    enum maxSize = Impl.sizeof;
 }
 
 ///
 @safe unittest
 {
+    struct Cat { int a, b, c; }
+
+    align(1) struct S {
+        long l;
+        ubyte b;
+    }
+
+    align(1) struct T {
+        ubyte b;
+        long l;
+    }
+
     static assert(maxSize!(int, long) == 8);
     static assert(maxSize!(bool, byte) == 1);
-
-    struct Cat { int a, b, c; }
     static assert(maxSize!(bool, Cat) == 12);
+    static assert(maxSize!(char) == 1);
+    static assert(maxSize!(char, short, ubyte) == 2);
+    static assert(maxSize!(char, long, ubyte) == 8);
+    static assert(maxSize!(long, S) == 9);
+    static assert(maxSize!(S, T) == 16);
```

Comment:
This line is causing a failure on systems with different alignment requirements where only `T.sizeof` is 12 rather than 16.

Reply:
So should we remove the line or case on some version?

Reply:
I was thinking more along the lines of using an inequality instead of a strict equality. But you'd have to think about it a bit more carefully to formulate it in a way that's meaningful.

Reply:
Would something like this work?

```d
import std.algorithm.comparison : max;
static assert(maxSize!(S, T) == max(S.sizeof, T.sizeof));
```

Reply:
@nordlow Can you please replace this line with the suggested one from @pbackus ?

Reply:
Done